workers/data_refinery_workers/processors/test_illumina.py

```diff
@@ -67,25 +69,108 @@ def prepare_illumina_job(organism):
+    og_file.absolute_file_path = os.path.realpath(path)
+    og_file.is_downloaded = True
+    og_file.save()
+
+    return og_file
+
+
+def _try_sanitizing_file(file: str) -> str:
+    pj = ProcessorJob()
+    pj.pipeline_applied = "ILLUMINA_TO_PCL"
+    pj.save()
+
+    og_file = _make_original_file_with_contents(file)
+
+    job_context = illumina._prepare_files({"job_id": pj.pk, "original_files": [og_file], "job": pj})
+    job_context = illumina._detect_encoding(job_context)
+    return illumina._sanitize_input_file(job_context)
+
+
+# Save this experiment separately (sans organism) because we need it for multiple tests
+GSE22427 = {
+    "source_filename": "ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE22nnn/GSE22427/suppl/GSE22427%5Fnon%2Dnormalized%2Etxt.gz",
+    "filename": "GSE22427_non-normalized.txt",
+    "absolute_file_path": "/home/user/data_store/raw/TEST/ILLUMINA/GSE22427_non-normalized.txt",
+    "samples": [
+        "LV-C&si-Control-1",
+        "LV-C&si-Control-2",
+        "LV-C&si-Control-3",
+        "LV-C&si-EZH2-1",
+        "LV-C&si-EZH2-2",
+        "LV-C&si-EZH2-3",
+        "LV-EZH2&si-EZH2-1",
+        "LV-EZH2&si-EZH2-2",
+        "LV-EZH2&si-EZH2-3",
+        "LV-T350A&si-EZH2-1",
+        "LV-T350A&si-EZH2-2",
+        "LV-T350A&si-EZH2-3",
+    ],
+}
+
+GSE22427_HEADER = [
+    "ID_REF",
+    "LV-C&si-Control-1",
+    "Detection Pval",
+    "LV-C&si-Control-2",
+    "Detection Pval",
+    "LV-C&si-Control-3",
+    "Detection Pval",
+    "LV-C&si-EZH2-1",
+    "Detection Pval",
+    "LV-C&si-EZH2-2",
+    "Detection Pval",
+    "LV-C&si-EZH2-3",
+    "Detection Pval",
+    "LV-EZH2&si-EZH2-1",
+    "Detection Pval",
+    "LV-EZH2&si-EZH2-2",
+    "Detection Pval",
+    "LV-EZH2&si- EZH2-3",
+    "Detection Pval",
+    "LV-T350A&si-EZH2-1",
+    "Detection Pval",
+    "LV- T350A&si-EZH2-2",
+    "Detection Pval",
+    "LV-T350A&si-EZH2-3",
+    "Detection Pval",
+]
+
+
+class IlluminaToPCLTestCase(TestCase, testing_utils.ProcessorJobTestCaseMixin):
     @tag("illumina")
     def test_illumina_to_pcl(self):
         """Most basic Illumina to PCL test"""
 
         organism = Organism(name="HOMO_SAPIENS", taxonomy_id=9606, is_scientific_name=True)
         organism.save()
 
-        job = prepare_illumina_job(organism)
+        job = prepare_illumina_job({**GSE22427, "organism": organism})
+
+        # Remove the title of one of the samples to make sure that we can still
+        # find its detection column using the description given as an annotation
+        sample = Sample.objects.get(title="LV-T350A&si-EZH2-3")
+        sample.title = "ignoreme_for_description"
+        sample.accession_code = "ignoreme_for_description"
+        sample.save()
+
         final_context = illumina.illumina_to_pcl(job.pk)
-        self.assertTrue(final_context["success"])
+        self.assertSucceeded(job)
+        # XXX: remove this but because the job failed the rest of this won't succeed
```

Comment:
I'm confused by this comment.

Reply:
Agh, I meant to remove these before marking as ready for review. Will do.